### PR TITLE
Authbranch

### PR DESCRIPTION
**Pull request by Benthusiast**
added env variables that are needed with google strategy

### CODE REVIEW

**Review by sniehus — APPROVED** [review]

**Review by colinwilliams91 — APPROVED** [review]

File: src/server/index.ts [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+  function(req, res) {
+    // Successful authentication, redirect home.
+    res.redirect('/');
+  });
 
 app.get('/', (req, res) => {
   res.sendFile(path.resolve(DIST_DIR, 'index.html'));
 });
 
+app.get('/typescript', (req, res) => {
```
Comment (copelka) [inline src/server/index.ts:37]: ?

File: src/server/index.ts [hunk 2]
```diff
@@ -1,22 +1,43 @@
...
+  secret: 'typescript sucks lolipops.',
+  resave: false,
+  saveUninitialized: false
+}));
+app.get('/auth/google',
+  passport.authenticate('google', { scope: ['profile', 'email'] }));
+
+app.get('/auth/google/callback', 
+  passport.authenticate('google', { failureRedirect: '/login' }),
+  function(req, res) {
```
Comment (copelka) [inline src/server/index.ts:28]: use ES6

File: src/server/auth/auth.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+const { GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET } = process.env;
+const User = {findOrCreate: (a, b)=> console.log('typescript is total garbage.', a, b)}  // DELETE ME WHEN USER SCHEMA IS PROPERLY IMPORTED
+passport.use(new GoogleStrategy({
+  clientID: GOOGLE_CLIENT_ID,
+  clientSecret: GOOGLE_CLIENT_SECRET,
+  callbackURL: "http://localhost:8080/auth/google/callback"
+},
+function(accessToken, refreshToken, profile, cb) {
+  User.findOrCreate({ googleId: profile.id }, function (err, user) {
```
Comment (copelka) [inline src/server/auth/auth.ts:14]: use ES6

File: src/server/auth/auth.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import passport from 'passport';
+import 'dotenv/config';
+//import GoogleStrategy from 'passport-google-oauth20';
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+const { GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET } = process.env;
+const User = {findOrCreate: (a, b)=> console.log('typescript is total garbage.', a, b)}  // DELETE ME WHEN USER SCHEMA IS PROPERLY IMPORTED
+passport.use(new GoogleStrategy({
+  clientID: GOOGLE_CLIENT_ID,
+  clientSecret: GOOGLE_CLIENT_SECRET,
+  callbackURL: "http://localhost:8080/auth/google/callback"
```
Comment (copelka) [inline src/server/auth/auth.ts:11]: put this in your env to make it easier for deployment

**Pull request merged**